build_projects/dotnet-host-build/PrepareTargets.cs

```diff
@@ -42,6 +42,17 @@ public class PrepareTargets
         public static BuildTargetResult Init(BuildTargetContext c)
         {
             var configEnv = Environment.GetEnvironmentVariable("CONFIGURATION");
+            string platformEnv = Environment.GetEnvironmentVariable("TARGETPLATFORM") ?? RuntimeEnvironment.OperatingSystemPlatform.ToString();
+            string targetRID = Environment.GetEnvironmentVariable("TARGETRID");
+            if (targetRID == null)
+            {
+                targetRID = RuntimeEnvironment.GetRuntimeIdentifier();
+                if (targetRID.StartsWith("win") && (targetRID.EndsWith("x86") || targetRID.EndsWith("x64")))
```

Comment:
Why do we need this check? Isnt targetRID already in the expected form? If not, what form is it in?